client/templates/layouts/core.js

```diff
@@ -12,45 +14,68 @@ class CoreLayoutBeesknees extends Component {
     structure: PropTypes.object
   }
 
+  getComponent(name) {
+    try {
+      if (name) {
+        return assertComponent(name);
+      }
+    } catch (e) {
+      // No-op
+    }
+    return null;
+  }
+
+  renderMain() {
+    const template = this.props.structure && this.props.structure.template;
+    const mainComponent = this.getComponent(template);
+    if (mainComponent) {
+      return React.createElement(mainComponent, {});
+    } else if (Template[template]) {
+      return (
+        <Blaze template={template}/>
+      );
+    }
+    return null;
+  }
+
   render() {
     const { layoutHeader, layoutFooter, template } = this.props.structure || {};
     const pageClassName = classnames({
       "page": true,
       "show-settings": this.props.actionViewIsOpen
     });
 
+    const headerComponent = layoutHeader && this.getComponent(layoutHeader);
+    const footerComponent = layoutFooter && this.getComponent(layoutFooter);
+
     return (
       <div className={pageClassName} id="reactionAppContainer">
-        { Template[layoutHeader] &&
-          <Blaze template={layoutHeader} className="reaction-navigation-header" />
-        }
-
-        <Blaze template="cartDrawer" className="reaction-cart-drawer" />
-
-        { Template[template] &&
-          <main>
-            <div className="rui beesknees">
-              <div className="bkdebug"><em>{"Bee's Knees layout"}</em></div>
-              <div className="bkdebug"><em>{"layoutHeader template:"}</em> {this.props.structure.layoutHeader}</div>
-              <div className="bkdebug"><em>{"layoutFooter template:"}</em> {this.props.structure.layoutFooter}</div>
-              <div className="bkdebug"><em>{"Main Template:"}</em> {this.props.structure.template}</div>
-            </div>
-            <Blaze template={template} />
-          </main>
-        }
-
-        { Template[layoutFooter] &&
-          <Blaze template={layoutFooter} className="reaction-navigation-footer footer-default" />
-        }
+
+        {headerComponent && React.createElement(headerComponent, {})}
+
+        <Blaze template="cartDrawer" className="reaction-cart-drawer"/>
+
+        <main>
+          <div className="rui beesknees">
```

Comment:
Do you think this debug header is still relevant? Is there a way we could give it more value or should we just remove it?

Reply:
@zenweasel  Can you elaborate a bit? Don't know what  `debug header` is exactly.

Reply:
I'd leave that as is. I find it useful as , well, debug bar..